insights/core/__init__.py

```diff
@@ -1161,6 +1181,29 @@ class parser :py:class:`TextFileOutput`.
     * A None value when there is no timestamp info in the log file
     """
 
+    def _handle_content(self, context):
+        self._lines = None
+        self._context = context
+        # pass None as a fake argument for compatibility
+        self.parse_content(None)
+
+    def parse_content(self, content=None):
+        """
+        Use all the defined scanners to search the log file, setting the
+        properties defined in the scanner.  However, do not pre-load the lines.
+
+        Keep None as a fake keyword argument for compatibility.
+        """
+        for scanner in self.scanners:
```

Comment:
@xiangce Because `_handle_content` calls `parse_content`, so it means `parse_content` is called when parsers are initialized.

So, it seams that files are loaded in the parser's initialization?

Reply:
Nope, it depends.
If no re-implementation of the `parse_content` in the child Parsers, this one will be called without any content being loaded.
Unless the "self.lines` is used (or called) in the re-implemented `parse_content`, the content will never be loaded in initialization.

Reply:
In the scanners, the `self.lines` are called. Such as:
```
@classmethod
    def token_scan(cls, result_key, token, check=all):
        """ 
        Define a property that is set to true if the given token is found in
        the log file.  Uses the __contains__ method of the log file.
        
        Parameters:
            result_key(str): the scanner key to register
            token(str or list): one or more strings to search for
            check(func): built-in function ``all`` or ``any`` applied to each line
        
        Returns:
            (bool): the property will contain True if a line contained (any
            or all) of the tokens given.
        """
        def _scan(self):
            search_by_expression = self._valid_search(token, check)
            return any(search_by_expression(l) for l in self.lines)
        
        cls.scan(result_key, _scan)

Reply:
Nope, the `*scan` functions just register the `result_key` and its scan function `_scan` to the `self.scanners`, but never calls it. The `_scan` will be only actually called when the registered `result_key` being called.
E.g.

```python
>>> MyTexter.token_scan('find_more', 'more')
>>> my_texter.find_more  # <--<< Here the `self.lines` will be finally called.
```

Reply:
Hm.., I might be wrong. Let me have a double check.

Reply:
@qinpingli - The former implementation is incorrect, I add a new `LazyLogFileOutput` which needs to load the content by hand explicitly. Let's have a talk about this one.